modules/Microsoft.Web/staticSites/config/readme.md

```diff
@@ -0,0 +1,43 @@
+# Static Site Config `[Microsoft.Web/staticSites/config]`
+
+This module deploys a Static Site Config.
+
+## Navigation
+
+- [Resource Types](#Resource-Types)
+- [Parameters](#Parameters)
+- [Outputs](#Outputs)
+
+## Resource Types
+
+| Resource Type | API Version |
+| :-- | :-- |
+| `Microsoft.Web/staticSites/config` | [2022-03-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Web/staticSites/config) |
```

Comment:
Same as other PR discussion. We're not linking the right API documentation here nor in other added children. Latest is quite older that the one used here and the implementation looks different as well. In the latest documented API, the name property only accepts functionappsettings as a value and kind is a separated property. In this implementation, name and kind properties seem to be mixed, which I cannot be sure about as I lack proper knowledge on this service. Anyway, I find inaccurate to link a different API documentation than the one used. 

I think we should reopen the discussion about what to do if the azure resource reference documentation is outdated or lacking (we also have modules for resources which are not there at all). Probably removing the link to the docs completely and documenting in the Wiki that if links are not there it means that no official documentation in ARR exists yet for the API version we are using.

Reply:
Agree. In many cases we cannot link to the documentation because it is frankly not up to date - or not existing at all. While I appreciate that we want to point users to documentation where possible, we should not forget that Bicep comes with its own documentation. I assume in this case, the original contributor @itpropro generatd the documentation (e.g. parameter metadata) from this one.

In any case, I'm also not too familiar with the service and would hence opt-out to refactor the code. There is a fair chance I'd 'improve' it to the worse. 

Tricky one.

Reply:
The latest API version is the one mentioned here, I don't really get the question. The only real source to rely on are the api specs that are documented [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/StaticSites.json). They clearly state that the latest stable version is 2022-03-01, which is also the one that fixes a lot of bugs with the older provider like not being able to link a user provided function app for example. 
I think we should always rely on the API specs as they are the source for every SDK, the Azure Portal, the REST APIs and the Docs.

Reply:
It is. However, like in the other PR I'm not convinced the Azure Resource Reference is the best reference, especially as it is not updated as often as it should be and should not restrict us. Maybe we can find out why this is the case and contact the team owning it. If not, maybe there is a better reference we can dig up.

Reply:
Thanks for clarifying @eriqua, I think the API was just released last month and thereby the Docs have not been auto generated. Good idea to ask the responsible team what their plan is on that.

Reply:
Opened a new issue #1735 to keep track of this discussion

Reply:
Version 2022-03-01 is now available in the Azure Resource Reference https://docs.microsoft.com/en-us/azure/templates/microsoft.web/2022-03-01/staticsites?pivots=deployment-language-bicep which would unblock this PR. On the bigger topic we still need to agree on how to proceed when misalignments would raise.

Reply:
I received some feedback and the latest API version should be available now. Once I finished my current efforts I'll wrap this PR up

Reply:
I just updated the ReadMes and triggered the pipeline again. Once (successfully) finished I think we can resolve this comment

Reply:
@mrmcake, @itpropro brace yourself, I'm afraid we still have the same issue for the config child module, for which latest still points to an outdated version. Apparently the resource reference does not have a unique documentation for the module under the latest API version. It instead provides two, based on the `kind` (although `name` is used instead of `kind`, differently than this implementation): https://learn.microsoft.com/en-us/azure/templates/microsoft.web/2022-03-01/staticsites/config-appsettings?pivots=deployment-language-bicep and https://learn.microsoft.com/en-us/azure/templates/microsoft.web/2022-03-01/staticsites/config-functionappsettings?pivots=deployment-language-bicep. I'm not sure how to proceed from here. In order not to postpone this PR any further, I could suggest to merge it but to keep the discussion open in issue #1735 and maybe call out this specific case with the PG. What's your take on that?

Reply:
It's similar with the Sites module I pressume. The problem with the Azure Resource Reference is anyways that it is updated and designed semi-manually - and hence is not always consistent, nor up to date